lib/backend/k8s/k8s.go

```diff
@@ -142,14 +142,20 @@ func NewKubeClient(kc *capi.KubeConfig) (*KubeClient, error) {
 	return kubeClient, nil
 }
 
+// EnsureInitialized checks that the necessary custom object definitions
+// exist in the backend. This usually passes when using etcd
+// as a backend but can often fail when using KDD as it relies
+// on various custom resources existing.
+// To ensure the datastore is initialized, this function checks that a
+// known custom resource is defined: FelixGlobalConfigSetting. It accomplishes this
+// by trying to set the ClusterType (an example of FelixGlobalConfigSetting).
 func (c *KubeClient) EnsureInitialized() error {
-	// Ensure ClusterType is set.
-	log.Info("Ensuring ClusterType is set")
+	log.Info("Ensuring datastore has been initialized.")
 	err := c.waitForClusterType()
 	if err != nil {
-		return fmt.Errorf("Failed to ensure ClusterType is set: %s", err)
+		return fmt.Errorf("Failed to ensure datastore has been initialized: \"%s\". Have you created custom resources and is calico authorized to access them?", err)
```

Comment:
It's a bit odd to see a `?` in an error message. How about something like: `return fmt.Errorf("Failed to ensure datastore has been initialized: \"%s\". Make sure the Custom Resource Definitions are created and Calico is authorized to access them", err)